osquery/tables/other/augeas.cpp

```diff
@@ -0,0 +1,50 @@
+#include <osquery/tables.h>
+
+#include <augeas.h>
+#include <boost/algorithm/string/join.hpp>
+
+namespace osquery {
+namespace tables {
+void match(QueryData &results, const augeas *aug, const char *pattern) {
+  char **matches = NULL;
+
+  int len = aug_match(aug, pattern, &matches);
+
+  for (int i = 0; i < len; i++) {
+    Row r;
+    const char *path = matches[i];
```

Comment:
If len > 0 then the matches array is filled. I can add an assertion that matches is not NULL if len is above zero if needed.